### PR TITLE
Drop forceUpdate from TimeSince; use state/props changes instead

### PR DESCRIPTION
**Pull request by benvinegar**
Improves performance of `TimeSince` component by removing the `forceUpdate` call we were using to trigger relative time updates.

Now we update relative time via `state.relative` – React handles the rest.

Makes a noticeable difference on the stream view, where rendering goes from ~25ms on 25 stream items down to ~2ms.

Note this is targeted against the react-0.14 branch.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/timeSince.jsx [hunk 1]
```diff
@@ -18,35 +27,46 @@ var TimeSince = React.createClass({
...
-    this.ticker = setInterval(this.ensureValidity, delay);
+  setRelativeDateTicker() {
+    const ONE_MINUTE_IN_MS = 3600;
+
+    this.ticker = setTimeout(() => {
+      this.setState({
+        relative: this.getRelativeDate()
+      });
+      this.setRelativeDateTicker();
+    }, ONE_MINUTE_IN_MS);
```
Comment (dcramer) [inline src/sentry/static/sentry/app/components/timeSince.jsx]: doesnt this mean 1s doesnt update until its 61s?

Comment (benvinegar) [inline src/sentry/static/sentry/app/components/timeSince.jsx]: Yeah, that's true (it was doing that before, too).

We should make the timeout according to the duration of the relative time. Follow-up?

Comment (dcramer) [inline src/sentry/static/sentry/app/components/timeSince.jsx]: yeah i guess thats true, dont know why it was 2600

**Comment by benvinegar** [comment]
Before:

image

After:

image

(Note these charts don't incorporate bar chart perf improvements from #2202.

File: src/sentry/static/sentry/app/components/timeSince.jsx (hunk 1 above)
Comment (benvinegar) [inline src/sentry/static/sentry/app/components/timeSince.jsx]: I kind of just assumed it was a mistake and 3600 was intended.

**Comment by benvinegar** [comment]
Closing because wrong target. See #2207.

**Pull request closed without merging**